### PR TITLE
Bandits & Experiments UI fast follows (v.2 of many?)

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
- New FF rule modal modal
- FF rule layout & button changes
- general UI fixes

image

image

### CODE REVIEW

**Review by lukesonnet — CHANGES_REQUESTED** [review]
Some comments below, but most feedback and bug reported in Slack.

File: packages/back-end/src/validators/features.ts [hunk 1]
```diff
@@ -121,7 +123,22 @@ const experimentRule = baseRule
...
+    regressionAdjustmentEnabled: z.boolean().optional(),
+    sequentialTestingEnabled: z.boolean().optional(),
+    sequentialTestingTuningParameter: z.number().optional(),
+    statsEngine: z.enum(statsEngines).optional(),
+    banditStage: z.enum(banditStageType).optional(),
+    banditStageDateStarted: z.date().optional(),
+    banditScheduleValue: z.number().optional(),
+    banditScheduleUnit: z.enum(["hours", "days"]).optional(),
+    banditBurnInValue: z.number().optional(),
+    banditBurnInUnit: z.enum(["hours", "days"]).optional(),
```
[the remark this answers is not in the document]
Comment (bryce-fitzsimons) [reply packages/back-end/src/validators/features.ts:141]: This is the validator for the experiment's POST event, but does not actually reflect what will be saved into the rule model. I agree, this is not great, but not sure I want to rewire all this within this PR.

File: packages/stats/gbstats/bayesian/bandits.py [hunk 2]
```diff
@@ -234,7 +234,7 @@ def compute_result(self) -> BanditResponse:
             for mn, s in zip(self.variation_means, np.sqrt(self.posterior_variance))
         ]
         min_n = 100 * self.num_variations
-        enough_data = sum(self.variation_counts) >= min_n
+        enough_data = self.current_sample_size >= min_n
```
[the remark this answers is not in the document]
Comment (bryce-fitzsimons) [reply packages/stats/gbstats/bayesian/bandits.py:237]: I think Luke Smith added this commit to my branch but might be good to double check...

File: packages/front-end/components/Archetype/AssignmentTester.tsx [hunk 3]
```diff
@@ -256,7 +256,7 @@ export default function AssignmentTester({ feature, version, project }: Props) {
                 Top-level: <span className="text-success">pass</span>.
               </span>{" "}
               <span>
-                Override rules:{" "}
+                Rules:{" "}
```
[the remark this answers is not in the document]
Comment (bryce-fitzsimons) [reply packages/front-end/components/Archetype/AssignmentTester.tsx:259]: I think I got all of the tsc ones...

File: packages/stats/gbstats/bayesian/bandits.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukesonnet) [reply packages/stats/gbstats/bayesian/bandits.py:237]: Oh shoot. Yeah I was happy to merge that to main but I see he had the base branch as yours here. That's great, thanks.

File: packages/back-end/src/validators/features.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukesonnet) [reply packages/back-end/src/validators/features.ts:141]: Ok, thanks for explaining.

**Review by lukesonnet — COMMENTED** [review]
resolving some comments, will do some testing later today of the new flow.

**Review by lukesonnet — APPROVED** [review]

**Pull request merged**